app/src/main/java/com/nutomic/syncthingandroid/activities/MainActivity.java

```diff
@@ -255,6 +257,17 @@ public void onCreate(Bundle savedInstanceState) {
         onNewIntent(getIntent());
     }
 
+    @Override
+    public void onResume() {
+        // Check if storage permission has been revoked at runtime.
+        if ((ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) !=
+            PackageManager.PERMISSION_GRANTED)) {
+            startActivity(new Intent(this, FirstStartActivity.class));
+            this.finish();
+        }
+        super.onResume();
```

Comment:
No idea, but should this be an else?

Reply:
According to the lint warning the super function is required to be called when overriding.

Reply:
I took a quick search: https://stackoverflow.com/a/12975029 does explain the android activity lifecycle musn't be prevented by a missing super call.
https://developer.android.com/guide/components/activities/activity-lifecycle has a nice diagram where the activity lifecycle is shown.